plugins/modules/openssl_privatekey_pipe.py

```diff
@@ -0,0 +1,112 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: openssl_privatekey_pipe
+short_description: Generate OpenSSL private keys without disk access
+version_added: 1.3.0
+description:
+    - This module allows one to (re)generate OpenSSL private keys without disk access.
+    - This allows to read and write keys to vaults without having to write intermediate versions to disk.
+    - Make sure to not write the result of this module into logs or to the console, as it contains private key data! Use the I(no_log) task option to be sure.
+    - Note that this module is implemented as an L(action plugin,https://docs.ansible.com/ansible/latest/plugins/action.html)
+      and will always be executed on the controller.
+author:
+    - Yanis Guenane (@Spredzy)
+    - Felix Fontein (@felixfontein)
+options:
+    content:
+        description:
+            - The current private key data.
+            - Needed for idempotency. If not provided, the module will always return a change, and all idempotence-related
+              options are ignored.
+        type: str
+    content_base64:
+        description:
+            - Set to C(true) if the content is base64 encoded.
+        type: bool
+        default: false
+    return_current_key:
+        description:
+            - Set to C(true) to return the current private key when the module did not generate a new one.
+            - Note that in case of check mode, when this option is not set to C(true), the module always returns the
+              current key (if it was provided) and Ansible will replace it by C(VALUE_SPECIFIED_IN_NO_LOG_PARAMETER).
+        type: bool
+        default: false
+extends_documentation_fragment:
+- community.crypto.module_privatekey
+seealso:
+- module: community.crypto.openssl_privatekey
+- module: community.crypto.openssl_privatekey_info
+'''
+
+EXAMPLES = r'''
+- name: Generate an OpenSSL private key with the default values (4096 bits, RSA)
+  community.crypto.openssl_privatekey_pipe:
+    path: /etc/ssl/private/ansible.com.pem
+  register: output
+  no_log: true  # make sure that private key data is not accidentally revealed in logs!
+- name: Show generated key
+  debug:
+    msg: "{{ output.privatekey }}"
+  # DO NOT OUTPUT KEY MATERIAL TO CONSOLE OR LOGS IN PRODUCTION!
+
+- name: Update sops-encrypted key with the community.sops collection
+  community.crypto.openssl_privatekey_pipe:
+    content: "{{ lookup('community.sops.sops', 'private_key.pem.sops') }}"
+    size: 2048
+  register: output
+  no_log: true  # make sure that private key data is not accidentally revealed in logs!
+- name: Update encrypted key when openssl_privatekey_pipe reported a change
+  community.sops.encrypt_sops:
+    path: private_key.pem.sops
+    content_text: output.privatekey
+  when: output is changed
```

Comment:
Ansible-lint would mark this as "This should be a handler". ;-)

I'm not sure what the impact running this every time instead of only on changes would be, maybe this `when` can just be removed?

Reply:
> Ansible-lint would mark this as "This should be a handler". ;-)

If that's true, this is one of the cases very ansible-lint is very wrong. Replacing such code by a handler is a really stupid idea. There are a lot of situations where handlers make sense, but also a lot where they don't. This is definitely one of the latter.

> I'm not sure what the impact running this every time instead of only on changes would be, maybe this `when` can just be removed?

It depends on whether you have set `return_current_key: true` in the `openssl_privatekey_pipe` call. If you didn't, `output.privatekey` would be `VALUE_SPECIFIED_IN_NO_LOG_PARAMETER` is `output is not changed`, and you really don't want to write that into your private key file.

Reply:
https://ansible-lint.readthedocs.io/en/latest/default_rules.html#tasks-that-run-when-changed-should-likely-be-handlers

I am also a bit torn about this linting rule, just something that I came across quite often and a pattern that I recognize now when I see it.

Reply:
I'd say that in maybe half of the use-cases of `when: x is changed` it is totally correct. But in the other half, it is just wrong. You can partially rescue things with `meta: flush_handlers`, but that can make things even messier.